bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoHandlerFactory.java

```diff
@@ -65,134 +51,72 @@
  * @author Gaël L'hopital - Initial contribution
  */
 @NonNullByDefault
-@Component(service = ThingHandlerFactory.class, configurationPid = "binding.netatmo")
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.netatmo", property = Constants.SERVICE_PID
+        + "=" + SERVICE_PID)
 public class NetatmoHandlerFactory extends BaseThingHandlerFactory {
     private final Logger logger = LoggerFactory.getLogger(NetatmoHandlerFactory.class);
-    private final Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
-    private final Map<ThingUID, ServiceRegistration<?>> webHookServiceRegs = new HashMap<>();
-    private final HttpService httpService;
-    private final NATherm1StateDescriptionProvider stateDescriptionProvider;
+
     private final TimeZoneProvider timeZoneProvider;
-    private final LocaleProvider localeProvider;
-    private final TranslationProvider translationProvider;
-    private boolean backgroundDiscovery;
+    private final NetatmoDescriptionProvider stateDescriptionProvider;
+    private final ApiBridge apiBridge;
+    private final NetatmoServlet webhookServlet;
 
     @Activate
-    public NetatmoHandlerFactory(final @Reference HttpService httpService,
-            final @Reference NATherm1StateDescriptionProvider stateDescriptionProvider,
-            final @Reference TimeZoneProvider timeZoneProvider, final @Reference LocaleProvider localeProvider,
-            final @Reference TranslationProvider translationProvider) {
-        this.httpService = httpService;
-        this.stateDescriptionProvider = stateDescriptionProvider;
+    public NetatmoHandlerFactory(@Reference HttpService httpService,
+            @Reference NetatmoDescriptionProvider stateDescriptionProvider,
+            @Reference TimeZoneProvider timeZoneProvider, @Reference ApiBridge apiBridge,
+            @Reference NetatmoServlet webhookServlet) {
+        this.webhookServlet = webhookServlet;
         this.timeZoneProvider = timeZoneProvider;
-        this.localeProvider = localeProvider;
-        this.translationProvider = translationProvider;
-    }
-
-    @Override
-    protected void activate(ComponentContext componentContext) {
-        super.activate(componentContext);
-        Dictionary<String, Object> properties = componentContext.getProperties();
-        Object property = properties.get("backgroundDiscovery");
-        if (property instanceof Boolean) {
-            backgroundDiscovery = ((Boolean) property).booleanValue();
-        } else {
-            backgroundDiscovery = false;
-        }
-        logger.debug("backgroundDiscovery {}", backgroundDiscovery);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.apiBridge = apiBridge;
     }
 
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
-        return (SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID));
+        return (thingTypeUID.getBindingId().equals(BINDING_ID));
```

Comment:
While this may work, I think it's wrong. There is a defined set of things that is supported and that should be used. Maybe the user could even create a wrong thing-type by just entering invalid thing-entries in textual configuration.

Reply:
Correct.

Reply:
Changed